### ISSUE TITLE
[Security] Request validation middleware preceds api-key middleware 

### ISSUE THREAD

**Issue by PatStLouis** [hyperledger/aries-cloudagent-python#2666]
When I have a protected api admin interface and I send an unauthenticated malformed payload to an endpoint I get a 422 response with a detailed error message. Fixing the request switches the response to a 401. I would expect a 401 regardless of the payload.

This could potentially leak information and may lead to vulnerabilities as unauthorized request data is being handled.

To reproduce:
```
curl -X 'POST' \
  'https://{agent_admin_endpoint}/present-proof/create-request' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{}'
```

This returns a 422:
```
{
  "proof_request": [
    "Missing data for required field."
  ]
}
```

**Comment by swcurran** [comment]
@esune FYI. Let’s discuss.

**Comment by amanji** [comment]
Does this just require a decorator to check authentication?

**Comment by esune** [comment]
This might just be a middleware execution order issue. @PatStLouis are you looking at this too while you are looking at the API validator improvements or do we need to tackle this separately?

**Comment by PatStLouis** [comment]
I also believe this to be the case. I haven't looked at this issue directly however I can keep that in mind while looking at the validator improvements.

**Comment by swcurran** [comment]
Fixed by #2796
